### ISSUE TITLE
Fullscreen menus

### ISSUE THREAD

**Issue by alexcoder04** [Skayo/TI-Nspire-Minesweeper#4]
All the menus for restarting the game, changing the settings and viewing highscores are implemented using the menus at the top of the screen (I guess it's called tool palette, but I'm not sure). What about full-screen graphical menus? Sounds like a nice enhancement, but I'm afraid that it's going to destroy the minimalist feeling...

However, I could imagine working on this feature.

**Comment by Skayo** [comment]
That certainly would be very cool! If it has the same "look" as the rest of the UI/Game it would definitely look a lot better than the simple menus.

You could work on this feature on a new branch an make a PR at the end so I can check it out!

Thank you for putting so much work into this!

**Comment by Skayo** [comment]
How are your UI/UX skills? Should I try to make a little mockup for the interface?

**Comment by alexcoder04** [comment]
I don't have very much experience so it would be nice if you have the time to.

**Comment by alexcoder04** [comment]
I've thought about the game structure before starting to code and this is what I've come up with:

![Minesweeper-Structure drawio](https://user-images.githubusercontent.com/70093931/147749841-642855bb-88b3-4e20-abde-494aafbcc6a0.png)

**Comment by Skayo** [comment]
Looks pretty good! The only thing I'd change is having the "Back to Main Menu" in the Pause Screen.

So in the running game you press ESC -> Pause Menu with
- Resume
- Settings
- Help
- Restart (with confirmation)
- Exit to Main Menu (or just "Exit") (with confirmation)

The Main Menu could also have a "Help" option.

Btw, I found this yesterday: https://tcrf.net/Proto:Minesweeper_(Windows,_1990)/Mine_2.9
It shows some images from the Original Game Prototype, with a settings menu.
I will probably base the mockups on the design they had back then with the buttons and so on.
Not sure if I have the time for it today tho.

**Comment by Skayo** [comment]
Another thing which could be nice is some kind of key remapping. So you can set your own buttons for moving the cursor, setting flags, etc.

Also I thought about some simple localization using the Locale API from TI Nspire. This would only require a table with  the translated strings for each language. I could provide a German translation for the game and it would just fallback to English if the user's language doesn't exist.

And don't worry I definitely want to help you with some features! I just hope my Lua knowledge is still up-to-date haha

I'll start with the mockups later today.

**Comment by Skayo** [comment]
Maybe you've already done it but if you are interested you can download the original minesweeper from [here](https://minesweepergame.com/download/windows-xp-minesweeper.php) and reverse engineer/compare the behavior and stuff.
I recently noticed that the smiley doesn't have the "scared" state while clicking on a tile

**Comment by alexcoder04** [comment]
In my opinion, these ideas are very nice and I enjoy tinkering with Lua so far, although I don't move forward very fast.
